src/Indice.AspNetCore.Identity/Features/PushNotifications/DevicesController.cs

```diff
@@ -105,7 +105,7 @@ internal class DevicesController : ControllerBase
             if(device?.IsPushNotificationsEnabled == true) {
                 return NoContent();
             }
-            await _pushNotificationService.Register(request.DeviceId.ToString(), request.PnsHandle, request.DevicePlatform, user.Id);
+            await _pushNotificationService.Register(request.DeviceId.ToString(), request.PnsHandle, request.DevicePlatform, user.Id,  request.Tags.ToArray());
```

Comment:
Since your "Tags" property on the `RegisterDeviceRequest` is not initialized with an empty list then you need to check your prop for nulls `request.Tags?.ToArray()`.

Reply:
Resolved